### PR TITLE
SubnetCalc: Updating to use the record template

### PR DESCRIPTION
**Pull request by mintsoft**
This basically looks the same; however it uses the template:

image

I found writing the tests for the structured answers really laborious; is there a short `.*` for the hash so that the structure only needs to be tested once?

### CODE REVIEW

**Comment by moollaza** [comment]
>I found writing the tests for the structured answers really laborious; is there a short .* for the hash so that the structure only needs to be tested once?

Unfortunately not, but I've raised the issue that we sorely need to revist the way tests are written for Goodies now that we have structured answers.

Will being it up again internally.

**Comment by moollaza** [comment]
@mintsoft actually, there's this method: https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/t/BPMToMs.t#L33

```
structured_answer => {
    data => -ANY-
    ...
}
```

Is that what you're hoping for?

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 1]
```diff
@@ -98,37 +98,43 @@ handle query => sub {
...
+	sub to_text
+	{
+		my ($data, $keys) = @_;
+		return join "\n", map {"$_: $data->{$_}";} @{$keys};
+	}
+	
+    return to_text(\%output, \@output_keys),
+		structured_answer => {
+			id => "Subnet Calculator",
+			description => "Calculates IPv4 Subnets to and from CIDR notation",
```
Comment (moollaza) [inline lib/DDG/Goodie/SubnetCalc.pm]: whoops, you also need to specify the `name` for this so the tab has a name.

`name => "Sysadmin"` or `name => "Subnet Calculator"` should work

Comment (mintsoft) [inline lib/DDG/Goodie/SubnetCalc.pm]: Ah I see, so the `data-=>title` is the subtitle, but the name is the instance answer tab name. I couldn't see any difference when it was set!

**Comment by mintsoft** [comment]
| @mintsoft actually, there's this method: https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/t/BPMToMs.t#L33

I'll give that a whack on the next one I do and see if it works

**Comment by mintsoft** [comment]
@moollaza All sorted: 
image

File: share/goodie/subnet_calc/subnet_calc.css [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.record .record__cell--key {
```
Comment (jagtalon) [inline share/goodie/subnet_calc/subnet_calc.css]: Namespace this too, por favor.

**Comment by mintsoft** [comment]
@jagtalon Done

**Comment by moollaza** [comment]
>Edit: Although I don't know what that grey lump is next to the tab:

No worries, it’s just a glitch in DuckPAN. Can ignore :+1:

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 3]
```diff
@@ -98,37 +97,44 @@ handle query => sub {
...
+	sub to_text
+	{
+		my ($data, $keys) = @_;
+		return join "\n", map {"$_: $data->{$_}";} @{$keys};
+	}
+	
+    return to_text(\%output, \@output_keys),
+		structured_answer => {
+			id => "subnet_calculator",
+			description => "Calculates IPv4 Subnets to and from CIDR notation",
```
Comment (moollaza) [inline lib/DDG/Goodie/SubnetCalc.pm]: Can remove this, it's not actually a property we look for. The descriptions will be handled by the IA pages.

**Comment by moollaza** [comment]
@mintsoft Almost done! One more small fix and this is good :+1:

**Comment by mintsoft** [comment]
@moollaza sorted

**Comment by mintsoft** [comment]
@moollaza ping

**Comment by moollaza** [comment]
@mintsoft sorry for the delay! Will get to this ASAP

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 4]
```diff
@@ -98,37 +97,43 @@ handle query => sub {
...
+	}
+	
+    return to_text(\%output, \@output_keys),
+		structured_answer => {
+			id => "subnet_calculator",
+			name => "Subnet Calculator",
+			templates => {
+				group => 'list',
+				options => {
+					content => 'record'
```
Comment (moollaza) [inline lib/DDG/Goodie/SubnetCalc.pm]: I think we should set `moreAt => 0` or supply a Wiki link for subnets. On mobile view I'm seeing a vertical separator where the "More at" link would be, so I believe either of those should fix things.

File: lib/DDG/Goodie/SubnetCalc.pm [hunk 5]
```diff
@@ -98,37 +97,43 @@ handle query => sub {
...
-	["Network", int_to_str($network) . "/$cidr (Class $class)"],
-	["Netmask", int_to_str($mask)],
-	["Specified", "$which_specified"],
+    my %output = (
+		"Network" => int_to_str($network) . "/$cidr (Class $class)",
+		"Netmask" => int_to_str($mask),
+		"Specified" => "$which_specified",
     );
 
+	my @output_keys = qw(Network Netmask Specified);
```
Comment (moollaza) [inline lib/DDG/Goodie/SubnetCalc.pm]: nit: I just realized these are tabs :scream: 

Please switch to spaces :smile:

File: lib/DDG/Goodie/SubnetCalc.pm (hunk 4 above)
Comment (moollaza) [inline lib/DDG/Goodie/SubnetCalc.pm]: Yup, setting it to `0` remove the separator on Mobile.

File: lib/DDG/Goodie/SubnetCalc.pm (hunk 5 above)
Comment (mintsoft) [inline lib/DDG/Goodie/SubnetCalc.pm]: Dang, I must have used the wrong editor

**Comment by mintsoft** [comment]
@moollaza changes made, I'm not in a position to run `duckpan server` at the moment, but it should be fine

**Comment by mintsoft** [comment]
@moollaza just tested it, all changes made and working!

**Comment by mintsoft** [comment]
related: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/1163

**Comment by moollaza** [comment]
@mintsoft LGTM :+1: 

Sorry for the delay! Thanks again for doing an awesome job!

**Pull request merged**